pandas/core/indexes/base.py

```diff
@@ -98,6 +98,7 @@ def cmp_method(self, other):
             # numpy will show a DeprecationWarning on invalid elementwise
             # comparisons, this will raise in the future
             with warnings.catch_warnings(record=True):
+                warnings.filterwarnings("ignore", "elementwise", FutureWarning)
```

Comment:
What's the point in using `warnings.filterwarnings` instead of `pytest.mark.filterwarnings`? Wondering if we shouldn't always use the latter for consistency

Reply:
laziness / not wanting to introduce copy-paste errors. New code should essentially always use marks, and I can make the changes here if you want. I mostly cleaned up Panel to remove these `catch_warnings` / `filterwarnings`, but introduced one copy-paste error in the process (caught before pushing I think), so I'm inclined to not make unnecessary changes.

Reply:
Makes sense. Maybe not a hard rule on this change but it would be helpful to use this instead in functions where there is a big diff